src/bsoncxx/test/bson_util_itoa.cpp

```diff
@@ -16,10 +16,23 @@
 
 #include <bsoncxx/private/itoa.hpp>
 
+#if defined(__CYGWIN__)
```

Comment:
Rather than going to all of this effort, how about just disabling the entire test if __CYGWIN__ is defined? I'm confident that if bson_itoa functions on all other platforms, that it will work just fine on cygwin. Additionally, Please add a comment with a link to the upstream bug so that someday in the future someone can notice that it is no longer required.